rotkehlchen/tests/api/test_historical_balances.py

```diff
@@ -826,3 +829,96 @@ def test_get_historical_prices_per_asset(
         str(START_TS): '10',
         str(START_TS + DAY_IN_SECONDS): '20',
     }
+
+    # test future timestamp exclusion
+    current_time = ts_now()
+    future_ts = current_time + DAY_IN_SECONDS
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'historicalpricesperassetresource',
+        ),
+        json={
+            'asset': 'BTC',
+            'from_timestamp': current_time - DAY_IN_SECONDS,
+            'to_timestamp': future_ts,  # Include timestamp in the future
+            'interval': DAY_IN_SECONDS,
+        },
+    )
+    result = assert_proper_sync_response_with_result(response)
+    assert str(future_ts) not in result['prices']
```

Comment:
maybe check that the max timestamp here is also current_time (or curernt time plus a bit)

Reply:
it won't be exactly current_time since timestamps are generated at fixed intervals (from_timestamp + interval*n).
the maximum timestamp would be the highest interval boundary that doesn't exceed current_time, not current_time itself.

or did i misunderstand what you meant?